Traceback (most recent call last):
  File "C:\Users\Acer\AppData\Local\Programs\Python\Python310\lib\site-packages\jupyter_cache\executors\utils.py", line 51, in single_nb_execution
    executenb(
  File "C:\Users\Acer\AppData\Local\Programs\Python\Python310\lib\site-packages\nbclient\client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "C:\Users\Acer\AppData\Local\Programs\Python\Python310\lib\site-packages\nbclient\util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "C:\Users\Acer\AppData\Local\Programs\Python\Python310\lib\site-packages\nbclient\util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "C:\Users\Acer\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 646, in run_until_complete
    return future.result()
  File "C:\Users\Acer\AppData\Local\Programs\Python\Python310\lib\site-packages\nbclient\client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "C:\Users\Acer\AppData\Local\Programs\Python\Python310\lib\site-packages\nbclient\client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\Acer\AppData\Local\Programs\Python\Python310\lib\site-packages\nbclient\client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
#let's plot the ordered mutual_info values per feature
mutual_info.sort_values(ascending=False).plot.bar(figsize=(20, 8))
------------------

[1;31m---------------------------------------------------------------------------[0m
[1;31mImportError[0m                               Traceback (most recent call last)
Cell [1;32mIn [15], line 2[0m
[0;32m      1[0m [38;5;66;03m#let's plot the ordered mutual_info values per feature[39;00m
[1;32m----> 2[0m [43mmutual_info[49m[38;5;241;43m.[39;49m[43msort_values[49m[43m([49m[43mascending[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m)[49m[38;5;241;43m.[39;49m[43mplot[49m[38;5;241;43m.[39;49m[43mbar[49m[43m([49m[43mfigsize[49m[38;5;241;43m=[39;49m[43m([49m[38;5;241;43m20[39;49m[43m,[49m[43m [49m[38;5;241;43m8[39;49m[43m)[49m[43m)[49m

File [1;32m~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\plotting\_core.py:1160[0m, in [0;36mPlotAccessor.bar[1;34m(self, x, y, **kwargs)[0m
[0;32m   1072[0m [38;5;129m@Appender[39m(
[0;32m   1073[0m     [38;5;124;03m"""[39;00m
[0;32m   1074[0m [38;5;124;03m    See Also[39;00m
[1;32m   (...)[0m
[0;32m   1149[0m [38;5;129m@Appender[39m(_bar_or_line_doc)
[0;32m   1150[0m [38;5;28;01mdef[39;00m [38;5;21mbar[39m([38;5;28mself[39m, x[38;5;241m=[39m[38;5;28;01mNone[39;00m, y[38;5;241m=[39m[38;5;28;01mNone[39;00m, [38;5;241m*[39m[38;5;241m*[39mkwargs) [38;5;241m-[39m[38;5;241m>[39m PlotAccessor:
[0;32m   1151[0m     [38;5;124;03m"""[39;00m
[0;32m   1152[0m [38;5;124;03m    Vertical bar plot.[39;00m
[0;32m   1153[0m 
[1;32m   (...)[0m
[0;32m   1158[0m [38;5;124;03m    other axis represents a measured value.[39;00m
[0;32m   1159[0m [38;5;124;03m    """[39;00m
[1;32m-> 1160[0m     [38;5;28;01mreturn[39;00m [38;5;28mself[39m(kind[38;5;241m=[39m[38;5;124m"[39m[38;5;124mbar[39m[38;5;124m"[39m, x[38;5;241m=[39mx, y[38;5;241m=[39my, [38;5;241m*[39m[38;5;241m*[39mkwargs)

File [1;32m~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\plotting\_core.py:921[0m, in [0;36mPlotAccessor.__call__[1;34m(self, *args, **kwargs)[0m
[0;32m    920[0m [38;5;28;01mdef[39;00m [38;5;21m__call__[39m([38;5;28mself[39m, [38;5;241m*[39margs, [38;5;241m*[39m[38;5;241m*[39mkwargs):
[1;32m--> 921[0m     plot_backend [38;5;241m=[39m [43m_get_plot_backend[49m[43m([49m[43mkwargs[49m[38;5;241;43m.[39;49m[43mpop[49m[43m([49m[38;5;124;43m"[39;49m[38;5;124;43mbackend[39;49m[38;5;124;43m"[39;49m[43m,[49m[43m [49m[38;5;28;43;01mNone[39;49;00m[43m)[49m[43m)[49m
[0;32m    923[0m     x, y, kind, kwargs [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_get_call_args(
[0;32m    924[0m         plot_backend[38;5;241m.[39m[38;5;18m__name__[39m, [38;5;28mself[39m[38;5;241m.[39m_parent, args, kwargs
[0;32m    925[0m     )
[0;32m    927[0m     kind [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39m_kind_aliases[38;5;241m.[39mget(kind, kind)

File [1;32m~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\plotting\_core.py:1908[0m, in [0;36m_get_plot_backend[1;34m(backend)[0m
[0;32m   1905[0m [38;5;28;01mif[39;00m backend_str [38;5;129;01min[39;00m _backends:
[0;32m   1906[0m     [38;5;28;01mreturn[39;00m _backends[backend_str]
[1;32m-> 1908[0m module [38;5;241m=[39m [43m_load_backend[49m[43m([49m[43mbackend_str[49m[43m)[49m
[0;32m   1909[0m _backends[backend_str] [38;5;241m=[39m module
[0;32m   1910[0m [38;5;28;01mreturn[39;00m module

File [1;32m~\AppData\Local\Programs\Python\Python310\lib\site-packages\pandas\plotting\_core.py:1839[0m, in [0;36m_load_backend[1;34m(backend)[0m
[0;32m   1837[0m         module [38;5;241m=[39m importlib[38;5;241m.[39mimport_module([38;5;124m"[39m[38;5;124mpandas.plotting._matplotlib[39m[38;5;124m"[39m)
[0;32m   1838[0m     [38;5;28;01mexcept[39;00m [38;5;167;01mImportError[39;00m:
[1;32m-> 1839[0m         [38;5;28;01mraise[39;00m [38;5;167;01mImportError[39;00m(
[0;32m   1840[0m             [38;5;124m"[39m[38;5;124mmatplotlib is required for plotting when the [39m[38;5;124m"[39m
[0;32m   1841[0m             [38;5;124m'[39m[38;5;124mdefault backend [39m[38;5;124m"[39m[38;5;124mmatplotlib[39m[38;5;124m"[39m[38;5;124m is selected.[39m[38;5;124m'[39m
[0;32m   1842[0m         ) [38;5;28;01mfrom[39;00m [38;5;28mNone[39m
[0;32m   1843[0m     [38;5;28;01mreturn[39;00m module
[0;32m   1845[0m found_backend [38;5;241m=[39m [38;5;28;01mFalse[39;00m

[1;31mImportError[0m: matplotlib is required for plotting when the default backend "matplotlib" is selected.
ImportError: matplotlib is required for plotting when the default backend "matplotlib" is selected.

